my_enumerables.rb

```diff
@@ -0,0 +1,203 @@
+module Enumerable
+  def my_each
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < length
+        modified_object << (yield self[i])
+        i += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
+
+  def my_each_with_index
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < to_a.length
+        modified_object << (yield self[i], i)
+        i += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
+
+  def my_select
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < length
+        modified_object << self[i] if (yield self[i]) == true
+        i += 1
+      end
+      modified_object
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length == length
+  end
+
+  def my_any?(pattern = nil)
+    false if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length.positive?
+  end
+
+  def my_none?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length.zero?
+  end
+
+  def my_count(*args)
+    count_array = []
```

Comment:
I have never seen any example of count which generated an array. Can you provide me an example that supports your implementation? All I know is, it returns numeric numbers. Please remove the array declaration.